include/swift/Threading/Impl/Pthreads.h

```diff
@@ -130,6 +130,29 @@ inline void lazy_mutex_unsafe_unlock(lazy_mutex_handle &handle) {
   (void)::pthread_mutex_unlock(&handle);
 }
 
+// .. Recursive mutex support ................................................
+
+using recursive_mutex_handle = ::pthread_mutex_t;
+
+inline void recursive_mutex_init(mutex_handle &handle, bool checked = false) {
```

Comment:
Similarly to the Linux implementation, these should probably take `recursive_mutex_handle` arguments.